### ISSUE TITLE
Jellyfin subtitles default to wrong language 

### ISSUE THREAD

**Issue by Kjw94** [jellyfin/jellyfin-androidtv#2883]
### Describe the bug

My subtitles are set to English (my preferred language), my subtitle mode is set to “always play” and I have “burn subtitles” turned on. 

The problem is that when I play a movie, the subtitles default to another language that is different than my preferred language (English) such as Arabic or Spanish. This means that I have to manually change the subtitle language every time I play a movie or tv episode. Can you please fix this?

### Application version

0.15.11

### Device information

Nvidia Shield TV (2017)

### Android version

Android 9.1.1 (33.2.0.157)

### Jellyfin server version

Unknown

**Comment by newton181** [comment]
Have you tried "Smart" instead "Always play"?

"Subtitles matching the language preference will be loaded when the audio is in a foreign language."

**Comment by zilexa** [comment]
I've had this issue for ages.. not even the language itself but for everything I play, I **always** have to enable/select subtitles every time I play an episode. Sometimes sometimes the language is selected but it still does not show.
With Kodi properly configured with Jellyfin Addon (Addon mode), I don't have this issue. 

--> I simply always want to show English subtitles (or whatever language I have selected as default). In my experience this is impossible to configure with Jellyfin.

"Smart" mode would be useless for me, because it would only load subs if the audio language is not English. That's why it's set to "Always Play".


(For audio, I have a similar issue btw, when the first/default audiotrack of a file is not English, for example Italian. Italian will be played. 
Server side, Settings > Playback > Preferred Audio language is set to English, the checkbox below has been disabled ("play default audio track regardless of language"). Not sure if this will help.)

**Comment by clstrickland** [comment]
I have this same issue. I am hard of hearing, so I always want subtitles on. I have at least English and Spanish subtitles available for everything. But, even though I set it to prefer English, it defaults to Spanish every time on every client.

**Comment by amiklosi** [comment]
Plus one for the option of **always** playing subtitles with the **preferred language**. I've gone through all permutations of the settings but it seems to be unachievable currently with Jellyfin.

**Comment by casrou** [comment]
Just chiming in to say I experience the same problem..

When playing a movie or TV it always defaults to some other language than English - even though English subtitles exists. See my settings below:

![Screenshot_20240220-213651](https://github.com/jellyfin/jellyfin-androidtv/assets/29134373/23626a11-2325-4b91-8ad4-351d1e8c950f)

**Comment by maxi1134** [comment]
Same problem here with chromecasts, It Would be great if this accessibility feature could be fixed.

**Comment by maxi1134** [comment]
Same problem still present  here with chromecasts, It Would be great if this accessibility feature could be fixed.

**Comment by JeroenNieuwenhuis** [comment]
Same here. Might be related to https://github.com/jellyfin/jellyfin/issues/11974

**Comment by jaimeferj** [comment]
I have been experiencing this same problem for years now. I would like to try and give it a try. Any suggestions on where to start looking?
Thanks

**Comment by kevpen** [comment]
Definitely a quality of life issue impacting me for sure

**Comment by maxi1134** [comment]
not stale

**Comment by MedAzizTousli** [comment]
Same problem here.

**Comment by nielsvanvelzen** [comment]
Subtitle handling in the app was (greatly) improved in the 0.18 release. The subtitle track that is used by default is **chosen by the Jellyfin server** and not by the client. As such this issue should be fixed or is actually an issue with the server.
Closing as solved.

**Comment by MedAzizTousli** [comment]
@nielsvanvelzen Thanks for answering.
Could you please tell me how to choose it by the Jellyfin server? I went through all the server settings, and [these](https://imgur.com/a/RJ4SAUo) 3 screenshots are the only places where I found "Subtitle" in Server settings, and they don't change the default subtitle behaviour for me.

**Comment by nielsvanvelzen** [comment]
Yes those are the subtitle options, together with metadata on the actual media files (like forced/default flags in mkv containers). If you already played an item before the server "locks in" the last used subtitle track.

**Comment by MedAzizTousli** [comment]
@nielsvanvelzen Any idea how to "force" embedded subtitles?

**Comment by nielsvanvelzen** [comment]
No I'm not that familiar with the selection logic. You can try our troubleshooting channels (forum/chat) listed at jellyfin.org/contact.
